include/rocksdb/utilities/transaction_db.h

```diff
@@ -158,6 +158,18 @@ struct DeadlockPath {
 
 class TransactionDB : public StackableDB {
  public:
+  // Optimized version of ::Write that makes use of skip_concurrency_control
+  // hint; if it is true it means that the applicatinn guratnees that the
+  // key-set in the write batch do not conflict with any concurrent transaction
+  // and hence the concurrency control mechanism could be skipped for this
+  // write.
+  using StackableDB::Write;
+  virtual Status Write(const WriteOptions& opts, WriteBatch* updates,
+                       bool skip_concurrency_control) {
```

Comment:
Does this broke StackableDB::Write()? How about move `skip_concurrency_control` to write options?

Reply:
This API is not visible to neither DB nor StackableDB. My thinking was that  concurrency_control is a transactional notion and we should avoid mixing its concepts to data structures that are used by DB.

Reply:
While `StackableDB::Write(opt, batch)` and `TransactionDB::Write(opt, batch, skip_cc)` don't hide each other, it creates confusion say if I cast a TransactionDB to a StackableDB and call Write(), it may be confusing which method I'm calling. It also make it impossible to extend `DB::Write()` with a third param of bool. I would prefer either change the name of this method (something like `WiteWithoutConcurrencyControl()`) or make the method signature completely different (something like `Write(WriteOptions, TransactionDBWriteOptions, WriteBatch)`).